core/commands/storage.go

```diff
@@ -654,33 +713,62 @@ the chunk and replies back to client for the next challenge step.`,
 		ss := sm.GetOrDefault(ssID)
 		ss.SetStatus(storage.InitStatus)
 		go controlSessionTimeout(ss)
-		chidInt64, err := strconv.ParseInt(channelID, 10, 64)
-		if err != nil {
-			sm.Remove(ssID, "")
-			return err
-		}
-		chID := &ledgerPb.ChannelID{
-			Id: chidInt64,
-		}
 		chunkInfo := ss.GetOrDefault(chunkHash)
-		chunkInfo.UpdateChunk(pid, n.Identity, chID, price)
+		chunkInfo.UpdateChunk(pid, n.Identity, price)
 		chunkInfo.SetState(storage.InitState)
 
-		// build connection with ledger
-		channelInfo, err := getChannelInfo(req.Context, channelID)
-		if err != nil {
-			sendSessionStatusChan(ss.SessionStatusChan, storage.InitStatus, false, err)
-			sm.Remove(ssID, "")
-			return err
-		}
-		log.Debug("Verified channel:", channelInfo)
-
 		sendSessionStatusChan(ss.SessionStatusChan, storage.InitStatus, true, nil)
-		go downloadChunkFromClient(chunkInfo, chunkHash, ssID, n, pid, req, env)
+		// review contract and send back to client
+		go reviewContractAndSign(chunkInfo, chunkHash, ssID, n, pid, req, env, []byte(halfSignedCont))
+		//go downloadChunkFromClient(chunkInfo, chunkHash, ssID, n, pid, req, env)
 		return nil
 	},
 }
 
+func reviewContractAndSign(chunkInfo *storage.Chunk, chunkHash string, ssID string, n *core.IpfsNode,
+	pid peer.ID, req *cmds.Request, env cmds.Environment, payerSig []byte) {
+	// TODO: Check if renter is paid, if so, download file
+	go downloadChunkFromClient(chunkInfo, chunkHash, ssID, n, pid, req, env)
```

Comment:
Shouldn't this happen at the end of this function? If review fails, there is no need to download chunk.